airflow/settings.py

```diff
@@ -63,13 +67,13 @@
 LOG_FORMAT = conf.get('core', 'log_format')
 SIMPLE_LOG_FORMAT = conf.get('core', 'simple_log_format')
 
-SQL_ALCHEMY_CONN = None
-DAGS_FOLDER = None
-PLUGINS_FOLDER = None
-LOGGING_CLASS_PATH = None
+SQL_ALCHEMY_CONN = None  # type: Optional[str]
+DAGS_FOLDER = None  # type: Optional[str]
+PLUGINS_FOLDER = None  # type: Optional[str]
```

Comment:
Do these need to be `Optional[]` -- by the after `import airflow` is done these will _always_ be set.

Reply:
mypy only enforces type check statically, this means any change that happens at runtime will not be visible to it. Given these globals (and hook in `test_gcp_cloud_build_hook`) are actually populated after the python interpreter starts running, the optional is required.

The optional type hint is basically added for this specific use case: this variable will be initialized as None, but will be changed at runtime.